Updated for saml login_hint support. (#2339)
FusionAuth/fusionauth-site: site/docs/v1/tech/core-concepts/applications.adoc

[field]#Require signature# [optional]#Optional#::
When enabled, all unsigned requests will be rejected.

> [field]#Login hint parameter# [optional]#Optional# [since]#Available since 1.47.0#::

Reviewer: Need to update the related image.

> The name of the login hint paramter provided by the service provider on an AuthnRequest. If this parameter is present, its value will be used to pre-populate the username field on the FusionAuth login form.

Reviewer: ```suggestion
The name of the login hint parameter provided by the service provider on an AuthnRequest. If this parameter is present, its value will be used to pre-populate the username field on the FusionAuth login form.
```

Reviewer: I'm a bit confused about how this works. Say I configure the `loginHintParameter` to be `userHint`. That means in the AuthNRequest, I'll get something like `<AuthNRequest userHint="dan@example.com" ...>` and FusionAuth will prepopulate the login form with `dan@example.com`? Is that the case? If so, would be awesome to have that explanation in this doc somewhere.

Author: Added the sentence 

_Note that this setting names an HTTP query parameter, not an element in the SAML AuthnRequest XML._

+
This field is required when [field]#Require signature# is enabled.

[field]#Enable login hint# [since]#Available since 1.47.0#::

> When enabled, FusionAuth will accept a username or email address as a login hint on a custom HTTP request parameter

Reviewer: ```suggestion
When enabled, FusionAuth will accept a username or email address as a login hint on a custom HTTP request parameter.
```

[field]#Login hint parameter name# [optional]#Optional# [default]#defaults to `login_hint`# [since]#Available since 1.47.0#::

> The name of the login hint parameter provided by the service provider on an AuthnRequest. If this parameter is present, its value will be used to pre-populate the username field on the FusionAuth login form.

Reviewer: I still don't quite get what this does. If I don't, not sure that other folks will. Let me write what I think you are saying:

If I set `application.samlv2Configuration.loginHintConfiguration.parameterName` to `login_name` on an application, then when I set up FusionAuth as an IdP, the SP can send a request which includes the parameter `login_name=richard@example.com` and FusionAuth will pre-populate `richard@example.com` into the login form.

Is that a fair explanation? If so, I'd drop it in here. If not, then please help me understand.

Author: Dan, this verbiage ok?

+
This field is required when [field]#Require signature# is enabled.

[field]#Enable login hint# [since]#Available since 1.47.0#::
When enabled, FusionAuth will accept a username or email address as a login hint on a custom HTTP request parameter.

[field]#Login hint parameter name# [optional]#Optional# [default]#defaults to `login_hint`# [since]#Available since 1.47.0#::
The name of the login hint parameter provided by the service provider on an AuthnRequest. If this parameter is present, its value will be used to pre-populate the username field on the FusionAuth login form.
+

> For example, say application.samlv2Configuration.loginHintConfiguration.parameterName is set to the value `login_name` on an application. Then when FusionAuth is set up as an IdP, the SP can send a request which includes the parameter login_name=richard@example.com, and FusionAuth will pre-populate richard@example.com into the login form.

Reviewer: ```suggestion
For example, suppose [field]#Enable login hint# is enabled and [field]#Login hint parameter name# has the value `login_name`. When FusionAuth is set up as an IdP, the SP can send a request which includes the parameter `login_name=richard@example.com`, and FusionAuth will pre-populate richard@example.com into the login form the end user sees.
```
